File: Oct2017/SolutionAddTwoNumbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class SolutionAddTwoNumbers(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        st1=[];
        st2=[];
        while l1 is not None or l2 is not None:
            if l1 is not None:
                st1.append(l1.val)
                l1=l1.next
            if l2 is not None:
                st2.append(l2.val)
                l2=l2.next
        remain=0
        tmp=None
        while st1 or st2 or remain !=0:
            n1=0
            n2=0
            if st1:
                n1=st1.pop()
            if st2:
                n2=st2.pop()
            sum=n1+n2+remain
            l3=ListNode(sum%10)
            remain=sum/10
            l3.next=tmp
            tmp=l3
        return tmp
            
        

File: 2017/Oct2017/SolutionAddTwoNumbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class SolutionAddTwoNumbers {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        Stack<Integer> a = new Stack<>();
        Stack<Integer> b = new Stack<>();
        while(l1 != null || l2 != null){
            if(l1 != null) {
                a.push(l1.val);
                l1=l1.next;
            }
            if(l2 != null){
                b.push(l2.val);
                l2=l2.next;
            }
            
        }
        int carry = 0;
        ListNode tmp = null;
        while((!a.isEmpty() ||!b.isEmpty()) || carry != 0){
            System.out.println("xxxxxxx carry :" + carry );
            int val1= (a.isEmpty())?(0):(a.pop());
            int val2= (b.isEmpty())?(0):(b.pop());    
            int sum = val1 + val2  + carry;
            int val = (sum < 10)?(sum):(sum %10);
            ListNode l3 = new ListNode(val);
            l3.next=tmp;
            tmp=l3;
            carry=(sum<10)?(0):(sum/10);
        }
        return tmp;   
    }
}